internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -655,6 +655,27 @@ func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
 	}
 }
 
+func (r *ClusterDeploymentsReconciler) getEncodedCACert(ctx context.Context,
+	log logrus.FieldLogger,
+	clusterDeployment *hivev1.ClusterDeployment,
+	clusterInstall *hiveext.AgentClusterInstall) (*string, error) {
+	caCertificateRef := clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference
+	if caCertificateRef != nil {
+		caSecret, err := getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: caCertificateRef.Namespace, Name: caCertificateRef.Name})
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting ca certificate secret")
+		}
+		caCertBytes, hasCACert := caSecret.Data[corev1.TLSCertKey]
+		encodedCACert := base64.StdEncoding.EncodeToString(caCertBytes)
```

Comment:
Shouldn't be after the if statement below?

Reply:
You are right, fixed it.